lua/roslyn/sln/utils.lua

```diff
@@ -82,61 +113,91 @@ function M.root(buffer)
         }
     end
 
+
     if broad_search then
         local git_root = vim.fs.root(buffer, ".git")
         local search_root = git_root and sln:match(git_root) and git_root or sln
 
-        local solutions = find_solutions(search_root)
+        local solutions, solution_filters = find_solutions(search_root)
 
         return {
             solutions = solutions,
+            solution_filters = solution_filters,
             projects = projects,
         }
     else
+        local slnf = targets.slnf_dir
+
         return {
             solutions = find_files_with_extension(sln, ".sln"),
+            solution_filters = slnf and find_files_with_extension(slnf, ".slnf"),
             projects = projects,
         }
     end
 end
 
----Tries to predict which solutions to use if we found some
----returning the potentially predicted solution
----Notifies the user if we still have multiple to choose from
+---Tries to predict which target to use if we found some
+---returning the potentially predicted target
 ---@param root RoslynNvimRootDir
----@return string?
-function M.predict_sln_file(root)
+---@return boolean multiple, string? predicted_target
+function M.predict_target(root)
     if not root.solutions then
-        return nil
+        return false, nil
     end
 
     local config = require("roslyn.config").get()
-    local solutions = vim.iter(root.solutions)
-        :filter(function(solution)
-            if config.ignore_sln and config.ignore_sln(solution) then
+
+    local title = "roslyn.nvim"
+    local old_path
+
+    if config.ignore_sln then
```

Comment:
I was actually thinking we could check this already in config.lua and just do something like this maybe:

```lua
if config.ignore_sln then
    vim.notify("...")
    if not config.ignore_target then
        config.ignore_target = config.ignore_sln
    end
end
```

Then this code would be a bit cleaner, and it would only print the warning when calling setup

Reply:
Good point. I have `lock_target = true`, which is why I hadn't thought of this.

Which also makes me realise that bc291551a637204088dc3a8d125eb424fbf0b41a could be  annoying if `lock_target = false`...

Reply:
I've applied the check and fallback to the original behaviour for `lock_target = false`.